api/v1alpha1/fenceagentsremediation_params.go

```diff
@@ -0,0 +1,389 @@
+}
+
+func (v *customValidator) getSpec(new runtime.Object) *FenceAgentsRemediationSpec {
+	var spec FenceAgentsRemediationSpec
+	if fart, isFart := new.(*FenceAgentsRemediationTemplate); isFart {
+		spec = fart.Spec.Template.Spec
+	} else {
+		far, _ := new.(*FenceAgentsRemediation)
+		spec = far.Spec
+	}
+	return &spec
+}
+
+// validateFenceAgentForNodes validates fence agent parameters for all the nodes defined in the spec
+// by creating temporary FAR CRs and using BuildFenceAgentParams
+func validateFenceAgentForNodes(ctx context.Context, k8sClient client.Client, namespace string, spec *FenceAgentsRemediationSpec) error {
+
+	// Check if template has any parameters at all
+	hasSharedParams := len(spec.SharedParameters) > 0
+	hasNodeParams := len(spec.NodeParameters) > 0
+	hasSecrets := spec.SharedSecretName != nil || spec.NodeSecretNames != nil
+
+	// If template has no parameters or secrets, template is considered invalid
+	if !hasSharedParams && !hasNodeParams && !hasSecrets {
+		err := errors.New(errorMissingParams)
+		paramsLog.Error(err, "Missing parameters")
+		return err
+	}
+
+	// Collect all unique node names from NodeParameters and NodeSecretNames
+	nodeNames := getNodeNamesFromSpec(spec)
+
+	// If no node-specific parameters, validate with shared parameters only, use a dummy placeholder for node name
+	if len(nodeNames) == 0 {
+		paramsLog.Info("validateFenceAgentForNodes no nodes found")
+		nodeNames = append(nodeNames, "temp-validation")
+	}
+	// Validate parameters for each node mentioned in NodeParameters
+	for _, nodeName := range nodeNames {
+		// Generate a temporary FAR CR from the template for this specific node
+		tempFAR := &FenceAgentsRemediation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      nodeName,
+				Namespace: namespace,
+			},
+			Spec: *spec,
+		}
+
+		// BuildFenceAgentParams handles secret collection and validation internally
+		_, _, err := BuildFenceAgentParams(ctx, k8sClient, tempFAR)
+		if err != nil {
+			// If BuildFenceAgentParams fails, return the validation error
+			return err
+		}
+	}
+	return nil
+}
+
+func getNodeNamesFromSpec(spec *FenceAgentsRemediationSpec) []string {
+	nodeNamesMap := make(map[string]bool)
+	for _, nodeMap := range spec.NodeParameters {
+		for nodeName := range nodeMap {
+			nodeNamesMap[string(nodeName)] = true
+		}
+	}
+	for nodeName := range spec.NodeSecretNames {
+		nodeNamesMap[string(nodeName)] = true
+	}
+
+	var nodeNames []string
+	for nodeName := range nodeNamesMap {
+		nodeNames = append(nodeNames, nodeName)
+	}
+
+	return nodeNames
+}
+
+// validateFenceAgentParams builds the fence agent parameters map with validation
+func validateFenceAgentParams(far *FenceAgentsRemediation, secretParams SecretParams) (map[ParameterName]string, error) {
+	nodeName := GetNodeName(far)
+	fenceAgentParams := make(map[ParameterName]string)
+
+	// Track parameter names for uniqueness validation
+	existingParams := make(map[ParameterName]bool)
```

Comment:
why do we need this? AFAICS it will always contain the same names as fenceAgentParams...?